cuda/src/memory/cuda/device_memory_resource.cpp

```diff
@@ -44,14 +39,9 @@ namespace vecmem::cuda {
         std::size_t,
         std::size_t
     ) {
-        int current_device = 0;
-
-        VECMEM_CUDA_ERROR_CHECK(cudaGetDevice(&current_device));
-        VECMEM_CUDA_ERROR_CHECK(cudaSetDevice(m_device));
+        details::select_device dev(m_device);
```

Comment:
I still don't think that this is necessary in this function... :thinking:

Reply:
I'm not sure, there doesn't seem to be a clear answer to this anywhere, but even [some of NVIDIA's own examples](https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/examples.html) set the device before calling free.

Reply:
Fine, you convinced me. (I thought the comment would be added when pressing "Resolve conversation"...)